AGENTS.md

```
@@ -1,472 +1,248 @@
+  → FileGenerationRequest (DTO with all parameters)
+  → ParallelFileGenerator.GenerateFilesAsync()
+  → Channel<FileWorkItem> (work distribution)
+  → Multiple producer tasks (generate FileData)
+  → Channel<FileData> (results)
+  → ZipArchiveService.CreateArchiveAsync()
+  → ZIP archive + load file
 ```
 
-### Testing
+**Key Components:**
+| Component | Purpose |
```

markdownlint-cli2 (0.18.1)

Tables should be surrounded by blank lines

(MD058, blanks-around-tables)